src/main/java/com/aws/greengrass/secretmanager/AWSSecretClient.java

```diff
@@ -59,24 +53,22 @@ public AWSSecretClient(LazyCredentialProvider credentialProvider, DeviceConfigur
      * @throws IOException If there is a network error
      */
     public GetSecretValueResponse getSecret(GetSecretValueRequest request) throws SecretManagerException, IOException {
-        String errorMsg = String.format("Exception occurred while fetching secrets "
-                        + "from AWSSecretsManager for secret: %s, version: %s, label: %s",
-                request.secretId(), request.versionId(), request.versionStage());
         try {
             validateInput(request);
-            BaseRetryableAccessor accessor = new BaseRetryableAccessor();
-            CrashableSupplier<GetSecretValueResponse, IOException> getSecretValueResponse =
+            CrashableSupplier<GetSecretValueResponse, Exception> getSecretValue =
                     () -> secretsManagerClient.getSecretValue(request);
-            GetSecretValueResponse response = accessor.retry(RETRY_COUNT, BACKOFF_MILLIS, getSecretValueResponse,
-                    new HashSet<>(Collections.singletonList(IOException.class)));
+            GetSecretValueResponse response = RetryUtils.runWithRetry(
+                    RetryUtils.RetryConfig.builder().maxAttempt(RETRY_COUNT)
+                            .retryableExceptions(Collections.singletonList(IOException.class)).build(),
+                    getSecretValue, "fetch-secret", LogManager.getLogger(AWSSecretClient.class));
             validateResponse(response);
             return response;
-        } catch (InternalServiceErrorException
-                | DecryptionFailureException
-                | ResourceNotFoundException
-                | InvalidParameterException
-                | InvalidRequestException
-                | IllegalArgumentException e) {
+        } catch (IOException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)